src/components/AccountPickerItem.jsx

```diff
@@ -36,37 +38,49 @@ export const AccountPickerItem = ({
         <span>{accountName}</span>
         {nameAndLoginDiffer && <small>{accountLogin}</small>}
       </div>
-      {hasError && (
-        <div
-          className={classNames(
-            styles['col-account-picker-button-status'],
-            styles['col-account-picker-button-status--error']
-          )}
-        >
-          <span>
-            {getErrorTitle(t, error, key => `connection.error.${key}.title`)}
-          </span>
-          <Icon icon="warning" />
+      {maintenance && (
+        <div className={styles['col-account-picker-button-status']}>
+          <span>{t('connector.maintenance')}</span>
         </div>
       )}
-      {hasUpdate &&
+      {!maintenance &&
+        hasError && (
+          <div
+            className={classNames(
+              styles['col-account-picker-button-status'],
+              styles['col-account-picker-button-status--error']
+            )}
+          >
+            <span>
+              {getErrorTitle(t, error, key => `connection.error.${key}.title`)}
+            </span>
+            <Icon icon="warning" />
+          </div>
+        )}
+      {!maintenance &&
```

Comment:
We are getting to much conditions here and the whole thing si beginning to be unmaintainable. We should have done something when the `&& !hasError` happened.

Anyway, as all those conditions are mutually exclusive, I suggest to use a `status`variable to avoid this pseudo "boolean flag" pattern.

Something like:
```js
const MAINTENANCE = 'MAINTENANCE'
const ERRORED = 'ERRORED'
const HAS_UPDATE = 'HAS_UPDATE'
const CONNECTED = 'CONNECTED'
const RUNNING = 'RUNNING'

const getStatus = ({ hasError, hasUpdate, isConnected, isRuning, maintenance }) => {
  if (maintenance) return MAINTENANCE
  if (hasError) return ERRORED
  if (hasUpdate) return HAS_UPDATE
  if (isConnected) return CONNECTED
  if (isRunning) return RUNNING
}
```
And another helper for the status rendering:
```jsx
const statusRenderers = {
  [MAINTENANCE]: () => (<Jsx />),
  [ERRORED]: () => (<Jsx />),
  [HAS_UPDATE]: () => (<Jsx />),
  [CONNECTED]: () => (<Jsx />),
  [RUNNING]: () => (<Jsx />),
}

const renderStatus = status => typeof statusRenderers[status] === 'function' ? statusRenderers[status]() : null
```

What do you think ? And @CPatchane ?

Reply:
Agreed about this way to do this, even if it's quite equivalent to component method 👍